neo/Network/P2P/TaskManager.cs

```diff
@@ -281,15 +286,11 @@ private void RequestTasks(TaskSession session)
                 var height = Blockchain.Singleton.Height;
                 if (state_height + 1 < height)
                 {
-                    var state = Blockchain.Singleton.GetStateRoot((uint)(state_height + 1));
-                    if (state is null || state.Flag == StateRootVerifyFlag.Unverified)
```

Comment:
Why remove this check?

Reply:
After every block persisted,  local state root will be calculated and persist into `StateRoots` with `Unverified` flag. Only check the block height is enough.
https://github.com/neo-project/neo/blob/95596f93a604f48d0a962b092b143cbdf0769bf5/neo/Ledger/Blockchain.cs#L656